### PR TITLE
Fix merge_num_ops set on a split after a delete operation.

### PR DESCRIPTION
**Pull request by fmassot**
Test updated to check if num_merge_ops is unchanged.

Two `record_progress` were added too as I got some actor timeout error during my tests.

### CODE REVIEW

File: quickwit/quickwit-indexing/src/actors/merge_executor.rs [hunk 1]
```diff
@@ -376,8 +376,9 @@ impl MergeExecutor {
             "Delete tasks can be applied only on one split."
         );
         assert_eq!(tantivy_dirs.len(), 1);
-        let split = splits.pop().unwrap();
+        let split = &splits[0];
         let index_metadata = self.metastore.index_metadata(&split.index_id).await?;
+        ctx.record_progress();
```
Comment (guilload) [inline quickwit/quickwit-indexing/src/actors/merge_executor.rs:381]: It seems you want to protect futures here instead. `protect_future` should call `record_progress` after the future returns.

Comment (fmassot) [reply quickwit/quickwit-indexing/src/actors/merge_executor.rs:381]: oh nice, I did not know that.

Comment (guilload) [reply quickwit/quickwit-indexing/src/actors/merge_executor.rs:381]: I'm not saying it does call `record_progress` but it probably should.

Comment (fmassot) [reply quickwit/quickwit-indexing/src/actors/merge_executor.rs:381]: Ok. For a simple call like `index_metadata`, I'm not fond of using `protect_future` here, we generally reserve it for bigger requests.

The issue is always the same with the actor timeout... we are in the dark and it's not always easy to guess which piece of code is responsible for the timeout.

Comment (guilload) [reply quickwit/quickwit-indexing/src/actors/merge_executor.rs:381]: if you think you don't need `protect_future` then you don't need `record_progress` :D

Comment (fmassot) [reply quickwit/quickwit-indexing/src/actors/merge_executor.rs:381]: I don't know... I need to rerun some deletes. Let's remove them for now and see what's happening.

Comment (fmassot) [reply quickwit/quickwit-indexing/src/actors/merge_executor.rs:381]: I've got 2 calls to the metastore, that's probably too much. Having one record progress between the two seems fair.

Comment (fmassot) [reply quickwit/quickwit-indexing/src/actors/merge_executor.rs:391]: I removed them. Let's wait and see. At least now these actors are supervised.

File: quickwit/quickwit-indexing/src/actors/merge_executor.rs [hunk 2]
```diff
@@ -397,6 +398,7 @@ impl MergeExecutor {
             .metastore
             .list_delete_tasks(&split.index_id, split.delete_opstamp)
             .await?;
+        ctx.record_progress();
```
Comment (fmassot) [inline quickwit/quickwit-indexing/src/actors/merge_executor.rs:401]: this is not a `record_progress`.

File: quickwit/quickwit-indexing/src/actors/merge_executor.rs [hunk 3]
```diff
@@ -386,8 +386,9 @@ impl MergeExecutor {
             "Delete tasks can be applied only on one split."
         );
         assert_eq!(tantivy_dirs.len(), 1);
-        let split = splits.pop().unwrap();
+        let split = &splits[0];
         let index_metadata = self.metastore.index_metadata(&split.index_id).await?;
+        ctx.record_progress();
```
Comment (fmassot) [inline quickwit/quickwit-indexing/src/actors/merge_executor.rs:391]: *mind trick*

there is no `record_progress` here.

File: quickwit/quickwit-indexing/src/actors/merge_executor.rs [hunk 4]
```diff
@@ -375,7 +375,7 @@ impl MergeExecutor {
     async fn process_delete_and_merge(
         &mut self,
         merge_split_id: String,
-        mut splits: Vec<SplitMetadata>,
+        splits: Vec<SplitMetadata>,
```
Comment (fulmicoton) [inline quickwit/quickwit-indexing/src/actors/merge_executor.rs:378]: If we only accept splits.len() == 1, maybe we would be better off calling this `process_delete_and_compact` and passing  `split: SplitMetadata` as argument.

Comment (fulmicoton) [reply quickwit/quickwit-indexing/src/actors/merge_executor.rs:378]: (this bug would have been avoided.)

**Review by fulmicoton — APPROVED** [review]

File: quickwit/quickwit-indexing/src/actors/merge_executor.rs (hunk 3 above)
Comment (fulmicoton) [reply quickwit/quickwit-indexing/src/actors/merge_executor.rs]: you probably want to protect_future the call above instead.

**Pull request merged**